### PR TITLE
Use java.time.Duration to configure timeouts

### PR DESCRIPTION
**Pull request by valfirst**
- [X] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

### CODE REVIEW

File: java/client/src/org/openqa/selenium/WebDriver.java [hunk 1]
```diff
@@ -274,9 +277,31 @@
...
+     * searching for multiple elements, the driver should poll the page until at least one element
+     * has been found or this timeout has expired.
+     * <p>
+     * Increasing the implicit wait timeout should be used judiciously as it will have an adverse
+     * effect on test run time, especially when used with slower location strategies like XPath.
+     *
+     * @param duration The duration to wait.
+     * @return A self reference.
+     */
+    Timeouts implicitlyWait(Duration duration);
```
Comment (shs96c) [inline java/client/src/org/openqa/selenium/WebDriver.java]: There should be a default implementation of this that delegates down to the old method (for now). People who've extended Selenium in their own projects will have a fighting chance of doing the upgrade to 4.0. In our own code, we should (of course!) always implement this method.

File: java/client/src/org/openqa/selenium/WebDriver.java [hunk 2]
```diff
@@ -286,17 +311,45 @@
...
     /**
+     * Sets the amount of time to wait for an asynchronous script to finish execution before
+     * throwing an error. If the timeout is negative, then the script will be allowed to run
+     * indefinitely.
+     *
+     * @param duration The timeout value.
+     * @return A self reference.
+     * @see JavascriptExecutor#executeAsyncScript(String, Object...)
+     */
+    Timeouts setScriptTimeout(Duration duration);
```
Comment (shs96c) [inline java/client/src/org/openqa/selenium/WebDriver.java]: As above.

File: java/client/src/org/openqa/selenium/WebDriver.java [hunk 3]
```diff
@@ -286,17 +311,45 @@
...
+    }
+
+    /**
+     * Sets the amount of time to wait for a page load to complete before throwing an error.
+     * If the timeout is negative, page loads can be indefinite.
+     *
+     * @param duration The timeout value.
+     * @return A Timeouts interface.
+     */
+    Timeouts pageLoadTimeout(Duration duration);
```
Comment (shs96c) [inline java/client/src/org/openqa/selenium/WebDriver.java]: As above.

**Review by shs96c — COMMENTED** [review]
I like the idea in general, but we need to be mindful of how people will upgrade. Most of your PR is updating tests to use the new APIs --- I can only imagine the scale of the task for many of our users. I also suspect that there are plenty of places where people have (re-) implemented the interfaces, so we should use default implementations that delegate to the old way of doing things to enable those folks to do an upgrade as cleanly as possible.

File: java/client/src/org/openqa/selenium/WebDriver.java (hunk 1 above)
Comment (valfirst) [reply java/client/src/org/openqa/selenium/WebDriver.java]: @shs96c done, I've changed the implementation approach of default interface methods

**Review by diemol — COMMENTED** [review]
@valfirst Could you please solve the conflicts so I can review again to merge this?

**Comment by valfirst** [comment]
@diemol I've resolved the conflicts, could you please review the changes?

File: java/client/src/org/openqa/selenium/remote/DriverCommand.java [hunk 4]
```diff
@@ -210,17 +210,14 @@ static CommandPayload SET_ALERT_VALUE(String keysToSend) {
   String SET_ALERT_CREDENTIALS = "setAlertCredentials";
 
   String SET_TIMEOUT = "setTimeout";
-  static CommandPayload SET_IMPLICIT_WAIT_TIMEOUT(long time, TimeUnit unit) {
-    return new CommandPayload(
-        SET_TIMEOUT, ImmutableMap.of("implicit", TimeUnit.MILLISECONDS.convert(time, unit)));
+  static CommandPayload SET_IMPLICIT_WAIT_TIMEOUT(Duration duration) {
```
Comment (diemol) [inline java/client/src/org/openqa/selenium/remote/DriverCommand.java:221]: Not sure if I am missing something, but shouldn't this interface use the same (or at least similar) approach as others? 

If I understand this change correctly, someone could be using `CommandPayload SET_IMPLICIT_WAIT_TIMEOUT(long time, TimeUnit unit)`, and if this change was merged and released, and they upgrade, their code would not compile. Is that right or maybe I am missing something?

Comment (diemol) [reply java/client/src/org/openqa/selenium/remote/DriverCommand.java:221]: I mean, someone could have implemented their own driver.

**Review by diemol — CHANGES_REQUESTED** [review]

File: java/client/src/org/openqa/selenium/remote/DriverCommand.java (hunk 4 above)
Comment (valfirst) [reply java/client/src/org/openqa/selenium/remote/DriverCommand.java:221]: @diemol you are right, the question is what should be considered as public API and what as internal API, anyway I'll add back old methods

Comment (diemol) [reply java/client/src/org/openqa/selenium/remote/DriverCommand.java:221]: Good point, but if the interface is public we cannot discard that some people have implemnted it.

**Review by diemol — APPROVED** [review]
Thank you @valfirst!

**Pull request merged**